### PR TITLE
Type hints for worksearch

### PR DESCRIPTION
**Pull request by cclauss**
Closes https://github.com/internetarchive/openlibrary/pull/6466#discussion_r886149529

Add a few type hints to one function and then [solve for None](https://github.com/internetarchive/openlibrary/pull/6578#issuecomment-1137181218).

### CODE REVIEW

File: openlibrary/plugins/worksearch/code.py [hunk 1]
```diff
@@ -1219,13 +1219,13 @@ def work_search(
...
         )
-        response = reply['response'] or ''
+        response = reply.get('response') or {}
     except (ValueError, OSError) as e:
         logger.error("Error in processing search API.")
         response = dict(start=0, numFound=0, docs=[], error=str(e))
 
     # backward compatibility
-    response['num_found'] = response['numFound']
+    response['num_found'] = response.get('numFound') or 0
```
Comment (cclauss) [inline openlibrary/plugins/worksearch/code.py:1228]: An alternative depending on how the caller reacts to 0 vs None.
```suggestion
    response['num_found'] = response.get('numFound') or None
```

File: openlibrary/plugins/worksearch/code.py [hunk 2]
```diff
@@ -1219,8 +1217,9 @@ def work_search(
             facet=facet,
             spellcheck_count=spellcheck_count,
         )
-        response = reply['response'] or ''
-    except (ValueError, OSError) as e:
+        assert reply, "Received None response from run_solr_query"
```
Comment (cdrini) [inline openlibrary/plugins/worksearch/code.py:1220]: I made this a bit more explicit. Getting `None` back should be treated as an exceptional circumstance, and should pass through the exception handling code.

**Review by cdrini — COMMENTED** [review]
Lgtm; tossing on testing for a bit

**Review by cdrini — APPROVED** [review]
Ok not going to test this too thoroughly :/ This code is such a nightmare. Why is `run_solr_query` returning `None` in the first place? It's also returning none even when solr provides a valid JSON response with an error message -_- But this should cause the error to still be logged in sentry, but not cause the actual code to error. What that means for down stream code? Who knows!

**Pull request merged**